### PR TITLE
AI Packages Generation

### PR DESCRIPTION
**Pull request by yshdb5**
### Description:  
This pull request introduces AI-powered generation of package proposals using the Gemini API, enhancing provider registration workflows with dynamic, context-aware suggestions. It also updates the user interface to facilitate AI interaction, updates tests, and ensures compatibility with updated dependencies.

- **What issue does it solve?** This PR simplifies the process of creating service packages by integrating AI-generated suggestions, improving efficiency and user experience during provider registration.  
- **What functionality does it add?** Providers can now generate service package proposals through AI. A new dialog window makes this feature easily accessible, while updated tests and dependencies ensure the application remains robust and up-to-date.  

### Related Issues:  
- #382 

### Changes Made:  
- **AI-Powered Package Proposals**:  
  - Implemented `PackagesAssistantViewModel` to interface with the Gemini API for generating package proposals.  
  - Enabled providers to input textual context for more tailored AI-generated package suggestions.  

- **ProviderRegistrationProfile UI Enhancements**:  
  - Updated `ProviderRegistrationProfile.kt` to include a dialog window where users can easily interact with the AI model to generate packages.  
  - Improved UI elements to facilitate smoother package creation and user interaction.  

- **Testing and Library Updates**:  
  - Updated and adapted the `EndToEndTestCreateProfile` to reflect the new AI-powered package generation process.  
  - Updated project libraries and dependencies to maintain compatibility.  

### How to Test:  
1. **AI Package Generation**:  
   - Register as a new user, choose to be a Provider.
   - Navigate to `ProviderRegistrationProfile.kt` and open the dialog for generating package proposals.  
   - Choose the number of packages needed and provide optional additional context for the generation.
   - Verify that packages are generated and correspond to the service chosen during registration.
2. **Package Confirmation**:  
   - Verify that users can confirm AI-generated packages.  
3. **Testing**:  
   - Run the `EndToEndTestCreateProfile` to ensure it validates the AI integration and related workflows.  

### Checklist:  
- [x] Code adheres to project coding standards.  
- [x] New and updated tests have been written and verified.  
- [x] Functionality has been manually tested.  
- [x] Code is free of known issues.  
- [x] Documentation has been updated where applicable.

### CODE REVIEW

**Review by ethanboren — APPROVED** [review]
### Overall Assessment:

Great work on integrating AI-powered package proposals into the provider registration workflow. The code is very clean, with well-written comments that effectively explain the functionality. The comprehensive tests you've added ensure that everything works correctly and significantly increase line coverage.

### Positive Aspects :

- **Clean Code and Comments:** The codebase is tidy, and the comments enhance readability and understanding.
- **Robust Testing:** The extensive tests provide confidence in the new features and improve overall reliability.
- **User Experience Enhancements:** The updated UI facilitates smooth interaction with the AI assistant, improving user experience.

### Suggestions for Improvement :
- **Confirmation Toast Notification:** Consider adding a toast message when users accept the AI-generated packages. This feedback will inform them of successful acceptance and prevent multiple regenerations, which could lead to duplicate packages in their feed.
- **Scrollable Package Views:** To ensure usability when package descriptions are lengthy, make the package display areas scrollable. This will allow users to always access the acceptance option regardless of description length.

### Conclusion :

Overall, you've done an excellent job. Implementing these minor suggestions will enhance functionality and user satisfaction even further.

**Review by hjamaly — APPROVED** [review]
**Strengths:**

- Innovative Feature: AI-powered package generation simplifies workflows and adds personalized suggestions through contextual inputs.
- Code Quality: Clean implementation with PackagesAssistantViewModel and robust dependency updates.
- Response Handling: Excellent work decoding Gemini API responses into actionable proposals.

**Suggestions for Improvement:**

- Dialog UI: The dialog could be more engaging. Consider:
  - Modern card-based layouts for better visual appeal.
  - Adding an overview of generated packages for smoother user interaction.

**Summary:** Fantastic functionality; refining the UI will elevate the user experience. Great work overall!

File: app/src/main/java/com/android/solvit/shared/model/packages/PackagesAssistantViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+          modelName = "gemini-1.5-flash",
+          apiKey = BuildConfig.GOOGLE_AI_API_KEY,
+          generationConfig =
+              generationConfig {
+                temperature = 0.15f
+                topK = 32
+                topP = 1f
+                maxOutputTokens = 4096
+                responseMimeType = "application/json"
+                responseSchema = schema
```
[the remark this answers is not in the document]
Comment (yshdb5) [reply app/src/main/java/com/android/solvit/shared/model/packages/PackagesAssistantViewModel.kt:106]: I added the UID to the Schema to allow an automatic conversion from the JSON to the Package Proposal. This eases the conversion even if we do not use the UID generated by the model.

**Comment by yshdb5** [comment]
> **Suggestions for Improvement:**
> 
> * Dialog UI: The dialog could be more engaging. Consider:
>   
>   * Modern card-based layouts for better visual appeal.
>   * Adding an overview of generated packages for smoother user interaction.

I will push an update with an overview of the generated packages!

**Pull request merged**